corehq/apps/app_manager/views/releases.py

```diff
@@ -344,6 +344,9 @@ def revert_to_copy(request, domain, app_id):
     """
     app = get_app(domain, app_id)
     copy = get_app(domain, request.POST['saved_app'])
+    if copy.get_doc_type() == 'LinkedApplication' and app.get_doc_type() == 'Application':
+        Application.convert_to_application(copy.id)
+        copy = get_app(domain, copy.id)
```

Comment:
You could move the `save()` call out of `convert_to_application` and avoid re-fetching the document - I believe as-is, this will modify the old app build, but I think you want to leave that be and modify the working copy.

Reply:
I'm a little confused what you mean by this. If I take `Application.get_db().save_doc(raw_linked_app_doc)` out of `convert_to_application`, then un-linking the apps won't work because the changes won't get saved. I understand not wanting to fetch the document twice, but what are you suggesting I replace that with?

Reply:
You're right, I didn't explain myself very clearly.  Our models frequently have methods which do some mutation to the object and then save it (the users models have a lot of this).  However, we do frequently want to do more than one transformation in a single operation, at which point the `save()` call becomes a hindrance.  `users` frequently uses an optional `commit` parameter to skip the save call for this purpose.  The alternative is to have that method do the transformation, then manually call `save()` outside of the method.

In this instance, the method in question also fetches the document, which makes it similarly hard to combine this method with other operations.  I'd suggest modifying the method to only do the transformation, making the caller responsible for fetching and saving the document (both places it's used already have the doc fetched).

This code fetches the doc, uses the ID to fetch the doc again, makes the transformation, saves the doc, and fetches the doc again to pick up the transformed version.

The other thing I was saying is that this modifies `copy`, which is an old app build - I think you want to keep `copy` intact, but remove the linkage fields before saving.  You could do that by removing those fields from `copy` (but without saving), or by removing them from `app` after reverting.